src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/MultiUserAKVTest.java

```diff
@@ -0,0 +1,643 @@
+        assertTrue(provider.verifyColumnMasterKeyMetadata(keyIDs[0], true, signature1));
+        assertEquals(1, getCacheSize(cmkMetadataCacheSizeGetterName, provider));
+
+        assertTrue(provider.verifyColumnMasterKeyMetadata(keyIDs[0], true, signature2));
+        assertEquals(1, getCacheSize(cmkMetadataCacheSizeGetterName, provider));
+
+        assertFalse(provider.verifyColumnMasterKeyMetadata(keyIDs[0], false, signatureWithoutEnclave));
+        assertEquals(1, getCacheSize(cmkMetadataCacheSizeGetterName, provider));
+    }
+
+    @Test
+    public void cekCacheEntryIsEvictedAfterTtlExpires() throws Exception {
+        SQLServerColumnEncryptionAzureKeyVaultProvider provider = createAKVProvider();
+
+        if (provider == null) {
+            fail(TestResource.getResource("R_AKVProviderNull"));
+        }
+
+        if (!isMasterKeyPathSetup) {
+            fail(TestResource.getResource("R_masterKeyPathNullOrEmpty"));
+        }
+
+        provider.setColumnEncryptionCacheTtl(Duration.ofSeconds(10));
+        byte[] plaintextKey = { 1, 2, 3 };
+        byte[] encryptedKey = provider.encryptColumnEncryptionKey(keyIDs[0], Constants.CEK_ALGORITHM, plaintextKey);
+
+        provider.decryptColumnEncryptionKey(keyIDs[0], Constants.CEK_ALGORITHM, encryptedKey);
+        assertTrue(cekCacheContainsKey(encryptedKey, provider));
+        assertEquals(1, getCacheSize(cekCacheSizeGetterName, provider));
+
+        Thread.sleep(TimeUnit.SECONDS.toMillis(11));
+        assertFalse(cekCacheContainsKey(encryptedKey, provider));
+        assertEquals(0, getCacheSize(cekCacheSizeGetterName, provider));
+    }
+
+    @Test
+    public void cekCacheShouldBeDisabledWhenAkvProviderIsRegisteredGlobally() throws Exception {
+        SQLServerColumnEncryptionAzureKeyVaultProvider provider = createAKVProvider();
+        
+        if (provider == null) {
+            fail(TestResource.getResource("R_AKVProviderNull"));
+        }
+
+        if (!isMasterKeyPathSetup) {
+            fail(TestResource.getResource("R_masterKeyPathNullOrEmpty"));
+        }
+
+        SQLServerConnection.unregisterColumnEncryptionKeyStoreProviders();
+        Map<String, SQLServerColumnEncryptionKeyStoreProvider> providerMap = new HashMap<String, SQLServerColumnEncryptionKeyStoreProvider>();
+        providerMap.put(Constants.AZURE_KEY_VAULT_NAME, provider);
+        SQLServerConnection.registerColumnEncryptionKeyStoreProviders(providerMap);
+
+        try (SQLServerConnection con = (SQLServerConnection) PrepUtil.getConnection(AETestConnectionString, AEInfo)) {
+            Field globalCustomCacheField = SQLServerConnection.class.getDeclaredField("globalCustomColumnEncryptionKeyStoreProviders");
+            globalCustomCacheField.setAccessible(true);
+
+            Map<String, SQLServerColumnEncryptionKeyStoreProvider> globalCacheFieldValue = (Map<String, SQLServerColumnEncryptionKeyStoreProvider>) globalCustomCacheField.get(con);

```

Comment:
Eclipse warning:
Multiple markers at this line
	- Type safety: Unchecked cast from Object to Map<String,
	 SQLServerColumnEncryptionKeyStoreProvider>
	- Type safety: Unchecked cast from Object to Map<String,
	 SQLServerColumnEncryptionKeyStoreProvider>

Reply:
Suppressed